sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -93,11 +93,49 @@ object ScalaNativePluginInternal {
                                      entry,
                                      verbose)
 
+      checkThatClangIsRecentEnough(clang)
+
       IO.createDirectory(target)
       unpackRtlib(classpath)
       compileNir(opts)
       compileLl(clang, target, appll, binary, clangOpts)
       Process(abs(binary)).!
     }
   )
-}
+
+  /**
+   * Tests whether the clang compiler is recent enough.
+   * <p>
+   * This is determined through looking up a built-in #define which is more reliable than testing
+   * for a specific version.
+   * <p>
+   * It might be better to use feature checking macros:
+   * http://clang.llvm.org/docs/LanguageExtensions.html#feature-checking-macros
+   */
+  private def checkThatClangIsRecentEnough(pathToClangBinary: File): Unit = {
+    val clang = maybeFile(pathToClangBinary)
+    val defines: Seq[String] = definesBuiltIn(clang).to[Seq].flatten
+    val clangIsRecentEnough = defines.contains("__DECIMAL_DIG__ __LDBL_DECIMAL_DIG__")
+
+    if (! clangIsRecentEnough) {
+      throw new MessageOnlyException(s"No recent installation of clang found at $pathToClangBinary.\nSee https://github.com/scala-native/scala-native/blob/master/docs/building.md for details.")
```

Comment:
Splitting this message into a few strings that are concatenated would help with reading the code. Generally try to stay within 80 chars limit. Scalafmt doesn't format sbt plugins at the moment unfortunately.